tests/expressions/tests.py

```diff
@@ -2333,6 +2333,34 @@ def test_resolve_output_field(self):
                 expr = Value(value)
                 self.assertIsInstance(expr.output_field, output_field_type)
 
+    def test_mixed_output_field_with_none_returns_field(self):
+        test_cases = [
+            (Value(3.14159) + Value(None), FloatField),
+            (Value(42) + Value(None), IntegerField),
+            (Value(Decimal("3.14")) + Value(None), DecimalField),
+        ]
+
+        for expression, result in test_cases:
+            with self.subTest(expression):
+                self.assertIsInstance(expression.output_field, result)
+
+    def test_mixed_output_field_with_none_returns_none(self):
+        test_cases = [
+            Value(None),
+            Value("a string") + Value(None),
+            Value(datetime.date(2019, 5, 15)) + Value(None),
+            Value(datetime.time(3, 16)) + Value(None),
+            Value(datetime.datetime(2019, 5, 15)) + Value(None),
+            Value(datetime.timedelta(1)) + Value(None),
+            Value(b"") + Value(None),
+            Value(uuid.uuid4()) + Value(None),
+        ]
+
+        for expression in test_cases:
+            with self.subTest(expression):
+                with self.assertRaises(FieldError):
```

Comment:
To update the FieldError to `OutputFieldIsNoneError` in https://github.com/django/django/pull/17890

Reply:
@charettes if I understand you correctly...

1. If we infer the output field (e.g. the nonsensical test cases here to trigger inference of `output_field` through `_connector_combinations`), we **_should not_** raise? 
This would be different from existing behaviour which was established here: https://github.com/django/django/pull/15271

2. If explicitly specifying the output field e.g. with `ExpressionWrapper(Value(...), Value(...), output_field=DateTimeField())`, similar to `Value(None, output_field=DateTimeField(null=True))`, this does not raise.

Reply:
Note to self: 
Need to understand why in `test_resolve_output_field_with_null`, `(IntegerField, Combinable.ADD, null),` for example will trigger the error message `Cannot infer type...` when it should be able to infer through `_connector_combinations`